### PR TITLE
Len and DeleteByKey

### PR DESCRIPTION
**Pull request by codebien**

### CODE REVIEW

File: atlas.go [hunk 1]
```diff
@@ -55,8 +77,10 @@ func (n *Node) Path() map[string]string {
 	return result
 }
 
-// AddLink adds the provided key value pair to the tree and returns the new Node that includes all key and values of
-// the parent node and the new key value provided. If a key matches it will have the new value provided here.
+// AddLink adds the key and value strings to the tree
+// and returns the new Node that includes all the key and values of
+// the parent Node plus the new key-value pair provided.
+// If a key matches it will have the new value provided here.
```
Comment (codebien) [inline atlas.go:83]: What do we mean here? Is it relevant from a caller's perspective?

Comment (mstoykov) [reply atlas.go:83]: I mean that if you have `a:2,b:3` and you want to add `b, 5` you will get `a:2,b:5` and not `a:2, b:3, b:5`  (if we imagine that you can have the same key multiple times)

**Review by mstoykov — APPROVED** [review]

File: atlas.go (hunk 1 above)
Comment (codebien) [reply atlas.go]: I've extended the documentation, check the latest version, please.

File: atlas.go [hunk 2]
```diff
@@ -55,8 +77,14 @@ func (n *Node) Path() map[string]string {
...
 }
 
-// AddLink adds the provided key value pair to the tree and returns the new Node that includes all key and values of
-// the parent node and the new key value provided. If a key matches it will have the new value provided here.
+// AddLink adds the key and value strings to the tree
+// and returns the new Node that includes all the key and values of
+// the parent Node plus the new key-value pair provided.
+//
+// If another pair with the same key and a different value already exists in the path
+// then the new Node will have the key replaced with the new provided pair.
```
Comment (mstoykov) [inline atlas.go:85]:
```suggestion
// then the new Node will have the key replaced with the new provided value from the pair.
```

File: atlas.go [hunk 3]
```diff
@@ -55,8 +77,14 @@ func (n *Node) Path() map[string]string {
...
+//   (e.g. adding keyA, val2 to a set of {keyA:val1, keyB:val3}
+//   will generate a final path of {keyA: val2, keyB: val3}).
```
Comment (mstoykov) [inline atlas.go:87]:
```suggestion
//   will return set {keyA: val2, keyB: val3}).
```

You are mixing `set` and `path` and using the same ways to talk about them. I find it better to just use the word set in this particular case. But we can also use `path`

File: .golangci.yml [hunk 4]
```diff
@@ -45,6 +48,13 @@ linters-settings:
   funlen:
     lines: 80
     statements: 60
+  goheader:
```
[the remark this answers is not in the document]
Comment (codebien) [reply .golangci.yml:51]: I would prefer to keep the k6's version as the main source, so we should open a PR there before and update here after.

[the remark this answers is not in the document]
Comment (codebien) [reply .golangci.yml]: Updated

**Review by mstoykov — APPROVED** [review]

**Pull request merged**